ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/ExecutionEngineChannelImpl.java

```diff
@@ -223,4 +224,34 @@ private ExecutionEngineChannelImpl(ExecutionEngineClient executionEngineClient,
                     slot,
                     executionPayloadHeader));
   }
+
+  @Override
+  public SafeFuture<ExecutionPayload> proposeBlindedBlock(
+      SignedBeaconBlock signedBlindedBeaconBlock) {
+    LOG.trace("calling proposeBlindedBlock(signedBlindedBeaconBlock={})", signedBlindedBeaconBlock);
+
+    checkArgument(
+        signedBlindedBeaconBlock
+            .getBeaconBlock()
+            .map(beaconBlock -> beaconBlock.getBody().isBlinded())
```

Comment:
You can use `signedBlinkedBeaconBlock.getMessage()` here rather than `getBeaconBlock()` to avoid dealing with the `Optional`.